.ci/publish-images.sh

```diff
@@ -37,14 +31,12 @@ if [ "${QUAY_PASSWORD}x" != "x" -a "${QUAY_USERNAME}x" != "x" ]; then
     echo "${QUAY_PASSWORD}" | docker login -u "${QUAY_USERNAME}" quay.io --password-stdin
 
     echo "Tagging ${BUILD_IMAGE} as quay.io/${BUILD_IMAGE}"
-    docker tag "${BUILD_IMAGE}" "quay.io/${BUILD_IMAGE}"
-
-    echo "Pushing 'quay.io/${BUILD_IMAGE}'"
-    docker push "quay.io/${BUILD_IMAGE}"
+    IMAGE_TAGS="${IMAGE_TAGS} --tag quay.io/${BUILD_IMAGE}"
 
     if [ "${MAJOR_MINOR_IMAGE}x" != "x" ]; then
-        echo "Pushing 'quay.io/${MAJOR_MINOR_IMAGE}' to quay.io"
-        docker tag "${MAJOR_MINOR_IMAGE}" "quay.io/${MAJOR_MINOR_IMAGE}"
-        docker push "quay.io/${MAJOR_MINOR_IMAGE}"
+        IMAGE_TAGS="${IMAGE_TAGS} --tag quay.io/${MAJOR_MINOR_IMAGE}"
     fi
 fi
+
+echo "Building with tags ${IMAGE_TAGS}"
+docker buildx build --build-arg=GOPROXY=${GOPROXY} --platform=linux/amd64,linux/arm64 ${IMAGE_TAGS} --file build/Dockerfile .
```

Comment:
docker buildx could build once and push with multi tags.

so just change here to concat `--tag`

Reply:
Do you know if podman/buildah support this? If so, might be a good idea to try them out instead, as they generally work better in CI environments.

Reply:
Looks like it doesn't, and I also can't seem to be able to get the regular Docker to work:

```
$ docker version
Client: Docker Engine - Community
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 17:03:54 2020
 OS/Arch:           linux/amd64
 Experimental:      false

$ docker buildx build --build-arg=GOPROXY=${GOPROXY} --platform=linux/amd64,linux/arm64 ${IMAGE_TAGS} --file build/Dockerfile .
unknown flag: --build-arg
See 'docker --help'.

$ docker buildx build --platform=linux/amd64,linux/arm64 ${IMAGE_TAGS} --file build/Dockerfile .
unknown flag: --platform
See 'docker --help'.

```

Reply:
buildah seem not ready for multi arch build. https://github.com/containers/buildah/issues/1590

local buildx setup could see https://github.com/docker/buildx#with-buildx-or-docker-1903

```
$ export DOCKER_BUILDKIT=1
$ docker build --platform=local -o . git://github.com/docker/buildx
$ mkdir -p ~/.docker/cli-plugins
$ mv buildx ~/.docker/cli-plugins/docker-buildx
```

Reply:
or just download binary from https://github.com/docker/buildx/releases
input `~/.docker/cli-plugins/docker-buildx`

Reply:
Instead of requiring a plugin, can't we just iterate over the arches we want? What's the advantage of using buildx?

Reply:
i think so. and need `qemu` to support `linux/arm64` in amd64 host by github workflow.

buildx is just a frontend for [buildkit](https://github.com/moby/buildkit) which is the next docker build tools.

first advantage is the multi arch build in same time and support [OCI image](https://github.com/opencontainers/image-spec), which put all image layers of multi arch under same tag.

then when could `docker pull jaegertracing/jaeger-operator:1.9.0`, in arm64 pull layers for arm64, and in amd64 pull layers for amd64.

and no need add some arch suffix `jaegertracing/jaeger-operator:1.9.0-amd64` && `jaegertracing/jaeger-operator:1.9.0-arm64`

Reply:
So, if we build something with `docker build --arch=amd64 jaegertracing/jaeger-operator:1.9.0` then `docker build --arch=arm64 jaegertracing/jaeger-operator:1.9.0`, would the second override the first ?

Reply:
Yes. in old way. this why docker team provide the buildkit.
in docker registry hub, with `/v2` apis. the oci image have a manifest wrap to store arch infos.

Reply:
or use `docker manifest create ` to merge them together https://docs.docker.com/engine/reference/commandline/manifest/

Reply:
I'm still unable to get it working. We'll probably need step-by-step instructions added to the CONTRIBUTING.md. Here's what I'm getting:

```console
$ /usr/bin/docker buildx build --platform=linux/amd64,linux/arm64 --tag quay.io/jpkroehling/jaeger-operator:v1.19.0-15-g4e2c8b9f --file build/Dockerfile .
WARN[0000] invalid non-bool value for BUILDX_NO_DEFAULT_LOAD:  
multiple platforms feature is currently not supported for docker driver. Please switch to a different driver (eg. "docker buildx create --use")

$ /usr/bin/docker buildx create --use
sleepy_shaw

$ /usr/bin/docker buildx build --platform linux/arm/v7,linux/arm64/v8,linux/amd64 --tag your-username/multiarch-example:buildx-latest . 
WARN[0000] invalid non-bool value for BUILDX_NO_DEFAULT_LOAD:  
WARN[0000] No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load 
[+] Building 6.1s (1/1) FINISHED                                                                                                                                                                                                              
 => ERROR [internal] booting buildkit                                                                                                                                                                                                    6.1s
 => => pulling image moby/buildkit:buildx-stable-1                                                                                                                                                                                       5.7s
 => => creating container buildx_buildkit_sleepy_shaw0                                                                                                                                                                                   0.4s
------
 > [internal] booting buildkit:
------
Error response from daemon: cgroups: cgroup mountpoint does not exist: unknown

```

Reply:
@jpkrohling could you print `docker info`?

Reply:
let's check step by step 

```
$ docker buildx create --use --name localbuilder
$ docker buildx inspect localbuilder --bootstrap

[+] Building 13.7s (1/1) FINISHED                                                                                                                 
 => [internal] booting buildkit                                                                                                             13.7s
 => => pulling image moby/buildkit:buildx-stable-1                                                                                          12.7s
 => => creating container buildx_buildkit_localbuilder0                                                                                      1.0s
Name:   localbuilder
Driver: docker-container

Nodes:
Name:      localbuilder0
Endpoint:  unix:///var/run/docker.sock
Status:    running
Platforms: linux/amd64, linux/arm64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
```
it should output like above. What your script output?

Reply:
seems issue of docker.
https://github.com/docker/for-linux/issues/219#issuecomment-692423012